File: VB.NET/CodeCarvings.Piczard.Examples.ExampleSet_A/design/controls/PageTitle.ascx.vb
' -------------------------------------------------------
' Piczard Examples | ExampleSet -A- VB.NET
' Copyright 2011-2012 Sergio Turolla - All Rights Reserved.
' Author: Sergio Turolla
' <codecarvings.com>
'  
' THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF 
' ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
' TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A 
' PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT 
' SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR 
' ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN 
' ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, 
' OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE 
' OR OTHER DEALINGS IN THE SOFTWARE.
' -------------------------------------------------------

Partial Class design_controls_PageTitle
    Inherits System.Web.UI.UserControl

    Protected Overrides Sub OnPreRender(ByVal e As System.EventArgs)
        Dim macroAreaTitle As String = ""
        Select Case Me.MacroAreaID
            Case 1
                macroAreaTitle = "Piczard: The Basics"
            Case 2
                macroAreaTitle = "Image manipulation"
            Case 3
                macroAreaTitle = "Web - PictureTrimmer"
            Case 4
                macroAreaTitle = "Customize Piczard"
            Case 5
                macroAreaTitle = "Web - Image Upload Demos"
            Case 6
                macroAreaTitle = "Third Party Plugins"
        End Select

        Me.litMacroAreaTitle.Text = HttpUtility.HtmlEncode(macroAreaTitle.ToUpper())
        Me.litMainTitle.Text = HttpUtility.HtmlEncode(Me.Title.ToUpper())

        MyBase.OnPreRender(e)
    End Sub

    Public Property MacroAreaID() As Integer
        Get
            Dim result As Object = Me.ViewState("MacroAreaID")
            If (result IsNot Nothing) Then
                Return DirectCast(result, Integer)
            Else
                Return 1
            End If
        End Get
        Set(ByVal value As Integer)
            Me.ViewState("MacroAreaID") = value
        End Set
    End Property

    Public Property Title() As String
        Get
            Dim result As Object = Me.ViewState("Title")
            If (result IsNot Nothing) Then
                Return DirectCast(result, String)
            Else
                Return ""
            End If
        End Get
        Set(ByVal value As String)
            Me.ViewState("Title") = value
        End Set
    End Property

End Class

File: CSharp/CodeCarvings.Piczard.Examples.ExampleSet_A/design/controls/PageTitle.ascx.cs
/*
 * Piczard Examples | ExampleSet -A- C#
 * Copyright 2011-2012 Sergio Turolla - All Rights Reserved.
 * Author: Sergio Turolla
 * <codecarvings.com>
 *  
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF 
 * ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
 * TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A 
 * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT 
 * SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR 
 * ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN 
 * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, 
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE 
 * OR OTHER DEALINGS IN THE SOFTWARE.
 */

using System;
using System.Collections.Generic;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;

public partial class design_controls_PageTitle : System.Web.UI.UserControl
{
    protected void Page_Load(object sender, EventArgs e)
    {

    }

    protected override void OnPreRender(EventArgs e)
    {
        string macroAreaTitle = "";
        switch (this.MacroAreaID)
        {
            case 1:
                macroAreaTitle = "Piczard: The Basics";
                break;
            case 2:
                macroAreaTitle = "Image manipulation";
                break;
            case 3:
                macroAreaTitle = "Web - PictureTrimmer";
                break;
            case 4:
                macroAreaTitle = "Customize Piczard";
                break;
            case 5:
                macroAreaTitle = "Web - Image Upload Demos";
                break;
            case 6:
                macroAreaTitle = "Third Party Plugins";
                break;
        }

        this.litMacroAreaTitle.Text = HttpUtility.HtmlEncode(macroAreaTitle.ToUpper());
        this.litMainTitle.Text = HttpUtility.HtmlEncode(this.Title.ToUpper());

        base.OnPreRender(e);
    }

    public int MacroAreaID
    {
        get
        {
            object result = this.ViewState["MacroAreaID"];
            if (result != null)
            {
                return (int)result;
            }
            else
            {
                return 1;
            }
        }
        set
        {
            this.ViewState["MacroAreaID"] = value;
        }
    }

    public string Title
    {
        get
        {
            object result = this.ViewState["Title"];
            if (result != null)
            {
                return (string)result;
            }
            else
            {
                return "";
            }
        }
        set
        {
            this.ViewState["Title"] = value;
        }
    }
}
